core/trino-main/src/main/java/io/trino/execution/buffer/OutputBuffers.java

```diff
@@ -56,33 +59,42 @@ else if (partitioningHandle.equals(FIXED_ARBITRARY_DISTRIBUTION)) {
         else {
             type = PARTITIONED;
         }
-        return new OutputBuffers(type, 0, false, ImmutableMap.of());
+        return new OutputBuffers(type, 0, false, ImmutableMap.of(), Optional.empty());
+    }
+
+    public static OutputBuffers createExternalExchangeOutputBuffers(ExchangeSinkInstanceHandle exchangeSinkInstanceHandle)
+    {
+        return new OutputBuffers(EXTERNAL, 0, true, ImmutableMap.of(), Optional.of(exchangeSinkInstanceHandle));
     }
 
     public enum BufferType
     {
         PARTITIONED,
         BROADCAST,
         ARBITRARY,
+        EXTERNAL,
```

Comment:
can you add an assertion to `withBuffer` and `withBuffers` that `bufferType != EXTERNAL`?

Reply:
It looks like it is not enforced by this class (e.g.: for partitioned output buffers)